sdk/resourcemanager/azure-resourcemanager-keyvault/src/test/java/com/azure/resourcemanager/keyvault/ManagedHsmTests.java

```diff
@@ -0,0 +1,140 @@
+        ManagedHsm managedHsm = createManagedHsm(mhsmName);
+
+        try {
+            // listByResourceGroups
+            PagedIterable<ManagedHsm> hsms = keyVaultManager.managedHsms()
+                .listByResourceGroup(rgName);
+            Assertions.assertTrue(hsms.stream().anyMatch(mhsm -> mhsm.name().equals(mhsmName)));
+
+            // getByResourceGroup
+            ManagedHsm hsm = keyVaultManager.managedHsms()
+                .getByResourceGroup(rgName, managedHsm.name());
+
+            // ManagedHsm properties
+            // The Azure Active Directory tenant ID that should be used for authenticating requests to the managed HSM pool.
+            String tenantId = hsm.tenantId();
+            Assertions.assertNotNull(tenantId);
+
+            ManagedHsmSku sku = hsm.sku();
+            Assertions.assertNotNull(sku);
+
+            // Array of initial administrators object ids for this managed hsm pool.
+            List<String> initialAdminObjectIds = hsm.initialAdminObjectIds();
+            Assertions.assertEquals(1, initialAdminObjectIds.size());
+
+            // The URI of the managed hsm pool for performing operations on keys.
+            String hsmUri = hsm.hsmUri();
+            Assertions.assertNotNull(hsmUri);
+
+            // Property to specify whether the 'soft delete' functionality is enabled for this managed HSM pool.
+            boolean softDelete = hsm.isSoftDeleteEnabled();
+            Assertions.assertTrue(softDelete);
+
+            // softDelete data retention days. It accepts >=7 and <=90.
+            Integer softDeleteRetentionDays = hsm.softDeleteRetentionDays();
+            Assertions.assertEquals(7, softDeleteRetentionDays);
+
+            // Property specifying whether protection against purge is enabled for this managed HSM pool.
+            boolean purgeProtectionEnabled = hsm.isPurgeProtectionEnabled();
+            Assertions.assertFalse(purgeProtectionEnabled);
+
+            // Rules governing the accessibility of the key vault from specific network locations.
+            MhsmNetworkRuleSet ruleSet = hsm.networkRuleSet();
+            Assertions.assertNotNull(ruleSet);
+
+            // Whether data plane traffic coming from public networks is allowed while private endpoint is enabled
+            PublicNetworkAccess publicNetworkAccess = hsm.publicNetworkAccess();
+            Assertions.assertEquals(PublicNetworkAccess.ENABLED, publicNetworkAccess);
+
+            // The scheduled purge date in UTC.
+            OffsetDateTime scheduledPurgeDate = hsm.scheduledPurgeDate();
+            Assertions.assertNull(scheduledPurgeDate);
+        } finally {
+            keyVaultManager.managedHsms().deleteById(managedHsm.id());
+            keyVaultManager.serviceClient().getManagedHsms().purgeDeleted(managedHsm.name(), managedHsm.regionName());
+        }
+    }
+
+    /*
+     * create or get managed hsm instance
+     */
+    // TODO(xiaofei) support managedHsm creation
```

Comment:
remove TODO?

Reply:
Yeah, currently it's serviceClient level code. Will remove it after we support it in convenience layer.